### PR TITLE
Change perms when moving annotations btwn groups

### LINKED ISSUE #2713: Post control not following scope change.
Go to a page, with scope as Public.
Select text.
Initiate an annotation.
Editor shows.
Now change scope to a group.
Post control shows "Only Me", instead of group name.

Video here:
https://drive.google.com/file/d/0B0yW-dSdEYAcZnZheFlEZm5yeG8/view?usp=drivesdk

Screenshot:
screenshot 2015-11-02 16 21 06

### PR DESCRIPTION
**Pull request by seanh**
Fixes #2713.

Needs test.

### CODE REVIEW

**Comment by seanh** [comment]
Test done.

File: h/static/scripts/directive/annotation.coffee [hunk 1]
```diff
@@ -429,7 +429,11 @@ AnnotationController = [
       # move any new annotations to the currently focused group when
       # switching groups. See GH #2689 for context
       if !model.id
-        model.group = groups.focused().id
+        debugger
```
Comment (nickstenning) [inline h/static/scripts/directive/annotation.coffee]: Probably not :)

**Comment by seanh** [comment]
Rebased on latest master with debugger statement removed

**Comment by seanh** [comment]
Ok this doesn't seem to fix the issue anymore...

**Comment by seanh** [comment]
This is ready now

**Comment by seanh** [comment]
This is ready now. It's a bit ugly (having to change both `model` and `@annotation` together) but it does actually seem to be necessary to preserve all the functionality, and at least we know that this pr isn't introducing any `model`/`@annotation` sync issues.

To be not horrible I think it needs this refactoring https://github.com/hypothesis/h/issues/2728

**Comment by nickstenning** [comment]
Good stuff. Sorting out AnnotationController needs to happen sooner rather than later. It's a disaster area right now.

**Comment by seanh** [comment]
Yep

**Pull request merged**